service/system.go

```diff
@@ -6,14 +6,48 @@ package service
 import (
 	"encoding/json"
 	"net/http"
-
-	"github.com/mattermost/mattermost/server/public/shared/mlog"
+	"time"
 
 	"github.com/prometheus/procfs"
+
+	"github.com/mattermost/mattermost/server/public/shared/mlog"
 )
 
 type SystemInfo struct {
-	Load procfs.LoadAvg `json:"load"`
+	CPULoad float64 `json:"cpu_load"`
+}
+
+func (s *Service) collectSystemInfo() {
+	// One second sampling interval.
+	ticker := time.NewTicker(time.Second)
+
+	var prevStat procfs.Stat
+	var prevTime time.Time
+
+	for {
+		select {
+		case currTime := <-ticker.C:
+			currStat, err := s.proc.Stat()
+
+			if err != nil {
+				s.log.Error("failed to get cpu stat", mlog.Err(err))
+				continue
+			}
+
+			idleDiff := currStat.CPUTotal.Idle - prevStat.CPUTotal.Idle
+
+			if !prevTime.IsZero() {
+				s.mut.Lock()
+				s.systemInfo.CPULoad = 1 / (idleDiff / currTime.Sub(prevTime).Seconds())
```

Comment:
This is to account for ticker skew/delay, or something else?

Reply:
Yes, it's normalizing it against the actual time spent.